packages/workbox-range-requests/src/lib/handle-range-request.js

```diff
@@ -0,0 +1,188 @@
+ *         });
+ *       })
+ *     );
+ *   }
+ * });
+ */
+
+/**
+ * @private
+ * @param {Object} input
+ * @param {string} input.rangeHeader A Range: header value.
+ * @return {Object} An object with `start` and `end` properties, reflecting
+ * the parsed value of the Range: header. If either the `start` or `end` are
+ * omitted, then `null` will be returned.
+ */
+export function parseRangeHeader({rangeHeader} = {}) {
+  isType({rangeHeader}, 'string');
+
+  const normalizedRangeHeader = rangeHeader.trim().toLowerCase();
+  if (!normalizedRangeHeader.startsWith('bytes=')) {
+    throw ErrorFactory.createError('unit-must-be-bytes');
+  }
+
+  if (normalizedRangeHeader.includes(',')) {
+    throw ErrorFactory.createError('single-range-only');
+  }
+
+  const rangeParts = /(\d*)-(\d*)/.exec(normalizedRangeHeader);
+  // We need either at least one of the start or end values.
+  if (rangeParts === null || !(rangeParts[1] || rangeParts[2])) {
+    throw ErrorFactory.createError('invalid-range-values');
+  }
+
+  return {
+    start: rangeParts[1] === '' ? null : Number(rangeParts[1]),
+    end: rangeParts[2] === '' ? null : Number(rangeParts[2]),
+  };
+}
+
+/**
+ * @private
+ * @param {Object} input
+ * @param {Blob} input.blob A source blob.
+ * @param {Number|null} input.start The offset to use as the start of the
+ * slice.
+ * @param {Number|null} input.end The offset to use as the end of the slice.
+ * @return {Object} An object with `start` and `end` properties, reflecting
+ * the effective boundaries to use given the size of the blob.
+ */
+export function calculateEffectiveBoundaries({blob, start, end} = {}) {
+  isInstance({blob}, Blob);
+  const blobSize = blob.size;
+
+  if (end > blobSize || start < 0) {
+    throw ErrorFactory.createError('range-not-satisfiable');
+  }
+
+  let effectiveStart;
+  let effectiveEnd;
+
+  if (start === null) {
+    effectiveStart = blobSize - end;
+    effectiveEnd = blobSize;
+  } else if (end === null) {
+    effectiveStart = start;
+    effectiveEnd = blobSize;
+  } else {
+    effectiveStart = start;
+    // Range values are inclusive, so add 1 to the value.
+    effectiveEnd = end + 1;
+  }
+
+  return {
+    start: effectiveStart,
+    end: effectiveEnd,
+  };
+}
+
+/**
+ * Given a `Request` and `Response` objects as input, this will return a
+ * promise for a new `Response`.
+ *
+ * @param {Object} input
+ * @param {Request} input.request A request, which should contain a Range:
+ * header.
+ * @param {Response} input.response An original response containing the full
+ * content.
+ * @return {Promise<Response>} Either a `206 Partial Content` response, with
+ * the response body set to the slice of content specified by the request's
+ * `Range:` header, or a `416 Range Not Satisfiable` response if the
+ * conditions of the `Range:` header can't be met.
+ */
+export async function handleRangeRequest({request, response} = {}) {
+  try {
+    isInstance({request}, Request);
```

Comment:
Should the isInstance checks be moved outside the try/catch to ensure the intended error message is seen by the developer?